graylog2-web-interface/src/views/types.d.ts

```diff
@@ -114,6 +114,9 @@ type BooleanField = BaseField & {
 
 type NumericField = BaseRequiredField & {
   type: 'numeric',
+  componentProps?: {
```

Comment:
I think we can flatten this. A `NumericField` can have an optional `step` key. It can be used inside the component rendering that field to pass that to the input. This way we have defined properties for the fields, which helps a lot when refactoring, because a user cannot just pass any prop to the field.